### PR TITLE
Set entity createdTimestamp and modifiedTimestamp in transaction handlers

### LINKED ISSUE #2131: Entity created and modify timestamps are null
**Detailed Description**
Entity created and modify timestamps are empty for all recent rows in mainnet.

**Actual Behavior**
View entity table on mainnet

**Expected Behavior**
Created and modified timestamps populated for all rows after the column was added.

After we address the root cause, we should run a migration to fix the data.

**Environment:**
- Java: OpenJDK 11.0.11
- OS: Ubuntu 18.04
- Version: v0.34.0

**Additional Context**

### PR DESCRIPTION
**Pull request by xin-hedera**
**Detailed description**:

- Add abstract base class AbstractEntityCrudTransactionHandler
- Make entity CRUD transaction handlers extend the base to set entity created/modified timestamp
- Add db migration to fill missing entity created/modified timestamp
- Rename Token\*TransactionsHandler to Token\*TransactionHandler

**Which issue(s) this PR fixes**:
Fixes #2131 

**Special notes for your reviewer**:

**Checklist**

- [ ] Documentation added
- [x] Tests updated

### CODE REVIEW

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Looks good.

I think you're missing a new migration file that would run the logic to set createdTimestamp and ModifiedTimestamps for those entities that were missed.
Unlike the initial run in V1.36.2 it can work on a subset of data

```sql
-- set created and modified timestamp based on entity creating or modifying transaction types
with entity__timestamp_map as (
    select entity_id, min(consensus_ns) as created_timestamp, max(consensus_ns) as modified_timestamp
    from transaction
    where result = 22
      and entity_id is not null
    group by entity_id
    order by entity_id, created_timestamp
)
update entity
set created_timestamp  = coalesce(created_timestamp, entity__timestamp_map.created_timestamp),
    modified_timestamp = entity__timestamp_map.modified_timestamp
from entity__timestamp_map
where id = entity__timestamp_map.entity_id and modified_timestamp is null;
```

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/transactionhandler/AbstractEntityCrudTransactionHandler.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import lombok.Getter;
+
+import com.hedera.mirror.importer.domain.Entity;
+import com.hedera.mirror.importer.domain.EntityId;
+import com.hedera.mirror.importer.parser.domain.RecordItem;
+
+abstract class AbstractEntityCrudTransactionHandler implements TransactionHandler {
+
+    @Getter
+    private final boolean createEntity;
```
Comment (Nana-EC) [inline hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/transactionhandler/AbstractEntityCrudTransactionHandler.java:32]: Should we make this an enum instead - `NONE`, `CREATE`, `UPDATE`, `DELETE`.
That might help in scaling and capturing intent

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/transactionhandler/AbstractEntityCrudTransactionHandler.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    @Override
+    public boolean updatesEntity() {
+        return true;
+    }
+
+    @Override
+    public void updateEntity(Entity entity, RecordItem recordItem) {
+        long consensusTimestamp = recordItem.getConsensusTimestamp();
+
+        if (createEntity) {
```
Comment (Nana-EC) [inline hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/transactionhandler/AbstractEntityCrudTransactionHandler.java:51]: If you adopt an enum then you could make this when `enum == Enum.CREATE`
And you'd also have a scenario for delete where you can set deleted to false.

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Looks good. 
Had an enum suggestion and a request for a migration test class

File: hedera-mirror-importer/src/main/resources/db/migration/v1/V1.40.0__entity_timestamp.sql [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+-- Fill missing entity created_timestamp and modified_timestamp
+-------------------
+
+-- set created and modified timestamp based on entity creating or modifying transaction types
+with entity__timestamp_map as (
+  select entity_id, min(consensus_ns) as created_timestamp, max(consensus_ns) as modified_timestamp
+  from transaction
+  where result = 22
+    and entity_id is not null
+    and consensus_ns > coalesce((
```
[the remark this answers is not in the document]
Comment (steven-sheehy) [reply hedera-mirror-importer/src/main/resources/db/migration/v1/V1.40.0__entity_timestamp.sql:11]: We really need to figure out an efficient way to run these migration tests. Right now every new migration test loads up a new application context, which is very slow. Seems like there should be a way to run flyway migrations ourselves one at a time and run its corresponding test before moving on to the next.

[the remark this answers is not in the document]
Comment (xin-hedera) [reply hedera-mirror-importer/src/main/resources/db/migration/v1/V1.40.0__entity_timestamp.sql:11]: sure. will add the migration test, also look into how to share the test application context between all the migration tests

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/transactionhandler/AbstractEntityCrudTransactionHandler.java (hunk 2 above)
Comment (xin-hedera) [reply hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/transactionhandler/AbstractEntityCrudTransactionHandler.java:51]: good suggestion will do.

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/EntityRecordItemListener.java [hunk 4]
```diff
@@ -444,20 +444,15 @@ private void insertEntityCreateOrUpdate(
             RecordItem recordItem, TransactionHandler transactionHandler, EntityId entityId) {
         Entity entity = entityId.toEntity();
         transactionHandler.updateEntity(entity, recordItem);
-        EntityId autoRenewAccount = transactionHandler.getAutoRenewAccount(recordItem);
-        if (!EntityId.isEmpty(autoRenewAccount)) {
-            entityListener.onEntity(autoRenewAccount.toEntity());
-            entity.setAutoRenewAccountId(autoRenewAccount);
+        if (!EntityId.isEmpty(entity.getAutoRenewAccountId())) {
+            entityListener.onEntity(entity.getAutoRenewAccountId().toEntity());
```
[the remark this answers is not in the document]
Comment (xin-hedera) [reply hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/EntityRecordItemListener.java:448]: sure

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/transactionhandler/AbstractEntityCrudTransactionHandler.java (hunk 1 above)
Comment (ijungmann) [reply hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/transactionhandler/AbstractEntityCrudTransactionHandler.java:32]: I like that idea, a `DELETE` option would further reduce some code.

**Review by ijungmann — COMMENTED** [review]
No additional feedback from what's already added, looks good.

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Review by ijungmann — APPROVED** [review]

**Review by Nana-EC — APPROVED** [review]
LGTM

**Pull request merged**